### PR TITLE
Enable react$ react$$ typings

### PR DESCRIPTION
**Pull request by tadashi0713**
## Proposed changes

Currently, types are missed in `react$` and `react$$`
(Types are enabled in `browser.react$` and `browser.react$$`)

Therefore, I enabled them

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by mgrybyk** [comment]
Are react$ assigned to global scope somewhere in the code?

File: packages/webdriverio/webdriverio.d.ts [hunk 1]
```diff
@@ -102,6 +108,8 @@ declare namespace WebdriverIOAsync {
 declare var browser: WebdriverIOAsync.BrowserObject;
 declare var $: $;
 declare var $$: $$;
+declare var react$: react$;
+declare var react$$: react$$;
```
Comment (christian-bromann) [inline packages/webdriverio/webdriverio.d.ts]: These are not declared as global variables, it is only for `$` and `$$`

Comment (tadashi0713) [reply packages/webdriverio/webdriverio.d.ts]: @christian-bromann Fixed, thanks

Comment (tadashi0713) [reply packages/webdriverio/webdriverio.d.ts]: ah, tests failed
Is that true that above are not necessary?

File: tests/typings/webdriverio/async.ts [hunk 2]
```diff
@@ -103,6 +103,10 @@ async function bar() {
     await reactElement.click()
     const reactElements = await reactWrapper.react$$('')
     await reactElements[0].click()
+    const reactElement2 = await react$('')
+    await reactElement2.click()
+    const reactElements2 = await react$$('')
+    await reactElements2[0].click()
```
Comment (christian-bromann) [inline tests/typings/webdriverio/async.ts]: this is not working
```suggestion
```

**Comment by christian-bromann** [comment]
ping @tadashi0713

**Comment by christian-bromann** [comment]
Any plans to move this forward?

**Comment by baruchvlz** [comment]
https://github.com/webdriverio/webdriverio/pull/4706 will make this obsolete as we're adding the `webdriverio-core.d.ts` to the package which already owns the definitions for the commands generated via JSDocs.

**Pull request closed without merging**